tools/iree-run-mlir-main.cc

```diff
@@ -44,561 +64,477 @@
+      auto stable_arg = std::make_unique<std::string>("--");
+      stable_arg->append(slice_arg);
+      temp_strings_.push_back(std::move(stable_arg));
+      out_args->push_back(temp_strings_.back()->data());
+    };
+    std::string_view sub_arg = prefixed_arg.substr(prefixed_arg.find(',') + 1);
+    for (;;) {
+      size_t comma_pos = sub_arg.find_first_of(',');
+      if (comma_pos == std::string_view::npos) break;
+      append_flag_string(sub_arg.substr(0, comma_pos));
+      sub_arg = sub_arg.substr(comma_pos + 1);
+    }
+    append_flag_string(sub_arg);
+  }
+
+  std::vector<std::unique_ptr<std::string>> temp_strings_;
+  std::vector<char*> runtime_args_;
+  std::vector<char*> compiler_args_;
+};
 
 }  // namespace
 
-extern "C" int main(int argc_llvm, char** argv_llvm) {
+extern "C" int main(int argc, char** argv) {
   IREE_TRACE_SCOPE0("iree-run-mlir");
 
-  mlir::DialectRegistry registry;
-  mlir::iree_compiler::registerAllDialects(registry);
-  mlir::iree_compiler::registerHALTargetBackends();
-  mlir::iree_compiler::registerVMTargets();
-  mlir::registerBuiltinDialectTranslation(registry);
-  mlir::registerLLVMDialectTranslation(registry);
-  // Make sure command line options are registered.
-  // Flag options structs (must resolve prior to CLI parsing).
-  (void)mlir::iree_compiler::BindingOptions::FromFlags::get();
-  (void)mlir::iree_compiler::InputDialectOptions::FromFlags::get();
-  (void)mlir::iree_compiler::HighLevelOptimizationOptions::FromFlags::get();
-  (void)mlir::iree_compiler::SchedulingOptions::FromFlags::get();
-  (void)mlir::iree_compiler::IREE::HAL::TargetOptions::FromFlags::get();
-  (void)mlir::iree_compiler::IREE::VM::TargetOptions::FromFlags::get();
-  (void)mlir::iree_compiler::IREE::VM::BytecodeTargetOptions::FromFlags::get();
-
-  // Register MLIRContext command-line options like
-  // -mlir-print-op-on-diagnostic.
-  mlir::registerMLIRContextCLOptions();
-  // Register assembly printer command-line options like
-  // -mlir-print-op-generic.
-  mlir::registerAsmPrinterCLOptions();
-  // Register pass manager command-line options like -mlir-print-ir-*.
-  mlir::registerPassManagerCLOptions();
-
-  // On Windows InitLLVM re-queries the command line from Windows directly and
-  // totally messes up the array.
-  llvm::setBugReportMsg(
-      "Please report issues to https://github.com/openxla/iree/issues and "
-      "include the crash backtrace.\n");
-  llvm::InitLLVM init_llvm(argc_llvm, argv_llvm);
-  llvm::cl::ParseCommandLineOptions(argc_llvm, argv_llvm);
-
-  // Consume all options after the positional filename and pass them to the IREE
-  // flag parser.
-  std::vector<char*> argv_iree = {argv_llvm[0]};
-  for (auto& run_arg : run_args_flag) {
-    if (run_arg == "--") continue;
-    argv_iree.push_back(const_cast<char*>(run_arg.c_str()));
+  // Initialize the compiler once on startup before using any other APIs.
+  ireeCompilerGlobalInitialize();
```

Comment:
I think something in here broke the "validate and publish release" workflow: https://github.com/openxla/iree/actions/runs/4829665331/jobs/8604961815#step:10:12

```
Run ./bin/iree-run-mlir --help
./bin/iree-run-mlir: error while loading shared libraries: libIREECompiler.so.0: cannot open shared object file: No such file or directory
```

https://github.com/openxla/iree/actions/workflows/validate_and_publish_release.yml
https://github.com/openxla/iree/blob/main/.github/workflows/validate_and_publish_release.yml